server/middleware/auth.ts

```diff
@@ -1,23 +1,30 @@
 import { Request, Response, NextFunction } from "express";
 import jwt from "jsonwebtoken";
+import { User } from "../models/user";
 
 const secret = process.env.JWT_SECRET as string;
 
-const auth = async (req: Request, res: Response, next: NextFunction) => {
-  try {
-    const token = req.signedCookies.token;
+function auth(userRepo: typeof User) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const token = req.cookies.token;
 
-    let decodedData;
-
-    if (token) {
-      decodedData = jwt.verify(token, secret) as jwt.JwtPayload;
-      res.locals.id = decodedData?.id;
+      if (token) {
+        const decodedData = jwt.verify(token, secret) as jwt.JwtPayload;
+        res.locals.user = await userRepo.findOne({
```

Comment:
Should we add a helper function in this file to pull the user off? It would help for handling the case where the locals value is missing so we don't need the extra code in the handlers.